### DISCUSSION TITLE
OwnCloud Infinite Scale - SelfHosted cloud

### DISCUSSION THREAD

**Discussion by xkpx64** [tteck/Proxmox#1869] [Ideas]
https://doc.owncloud.com/ocis/next/
Can anyone help me to build installer script for this awesome platform.

From docs:
$ sudo wget -O /usr/local/bin/ocis https://download.owncloud.com/ocis/ocis/stable/4.0.1/ocis-4.0.1-linux-amd64
$ sudo chmod +x /usr/local/bin/ocis
$ ocis init
$ OCIS_INSECURE=true OCIS_URL=https://mydata.net PROXY_HTTP_ADDR=0.0.0.0:9200 ocis server

**Comment by tteck** [comment]
Have you tried the TurnKey ownCloud installation?

**Reply by HerzogVonWiesel** [reply]
OwnCloud Infinite Scale is a new implementation that uses a whole other backend and is quite completely detached from the normal OwnCloud; It's way better at syncing files though, which is why it would be awesome to have. Any ideas you could give to how to write the script? :) Thank you so much!

https://doc.owncloud.com/ocis/next/deployment/

**Reply by tteck** [reply]
I'd start [here](https://doc.owncloud.com/ocis/next/deployment/binary/binary-setup.html).

**Reply by HerzogVonWiesel** [reply]
> I'd start [here](https://doc.owncloud.com/ocis/next/deployment/binary/binary-setup.html).

Thank you! Seems that im not up for the task with my current skills. Would love your rendition of a script if you ever find the time!

**Comment by xkpx64** [comment]
Yes, but its old version that uses php ( https://www.turnkeylinux.org/owncloud ) 
New version is based on GoLang microservices, really small and effective

**Comment by Vakrehus** [comment]
Is anyone coming to the login screen?

Server is running, npm set up, in my browser at "cloud.domain.com" only the loading spinner appears - no login screen, no error in console.

SOLVED: reinstall with baremetal

**Reply by nepumax** [reply]
Same problem.

Can you elaborate, what you did exactly to solve the neverending circle?

I installed Debian in LXC and followed the step-by-step guide for baremetal with systemd.

**Reply by geirskogul** [reply]
I know it's been a bit but I solved it by grabbing a cert with nginx proxy manager through the built-in letsencrypt setup, along with having my domain already set up and pointed to it.  Having the domain not set up and trying to access it entirely locally, as well as trying to run it with http only without a cert, killed it.  Even though I'm using http for the moment, it needed the cert for the initial login to pop up.

**Comment by jeremyjoslin** [comment]
Adding interest here for OCIS. It's a completely different product than the Turnkey distro.

**Comment by volpex-industries** [comment]
Yeah, I've bean poking at this for a few hours and have had nothing but problems. An oCIS installer script for this would be really appreciated.

**Comment by johnsturgeon** [comment]
Coming along to bump this as a request for a new script Specifically for OCIS and not ownCloud 10 (different products).

**Comment by Jmanko16** [comment]
Agree.  I'm interested in using this to replace synology drive, but I can't get it setup under Proxmox.  Would love a script.

**Comment by geirskogul** [comment]
I am requesting this myself.  Nextcloud and OG owncloud are super bloated compared to OCIS.  I would enjoy it a lot.
